### PR TITLE
Fix/restaurant details

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
## Summary

This pull request adds several new features and improvements to the TableTop app. It includes the following changes:

- Adds functionality to update restaurant venue details.
- Adds functionality to update venue images.
- Starts with the deletion of restaurant facilities.
- Refactors functionality to add a facility per venue.

## Details

This commit is a significant update to the TableTop app, as it introduces several new features and improvements. The commit adds the ability to delete a restaurant admin, which is a crucial functionality for managing the app. It also adds a second language to the remove admin feature, making it more accessible to non-English speaking users.

In addition, the commit adds functionality to update restaurant venue details and venue images, which are important features for maintaining accurate and up-to-date information about the restaurants. The commit also starts with the deletion of restaurant facilities, which will be completed in future updates.

The commit also includes a refactor of the functionality to add a facility per venue, which will improve the performance and maintainability of the app. Additionally, the commit fixes indentation issues in the edit modal, which will enhance the readability of the code.

Finally, the commit merges the `feature/removeResAdmin` branch into `TableTop-FE`, which brings all the changes made in that branch to the main codebase. It also pulls the latest changes from the `main` branch, ensuring that the code is up-to-date with the latest developments.

### CODE REVIEW

File: src/assets/languages/languageSelector.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ const languageSelector = (lang, str) => {
     restaurantOptionsReview: "REVIEWS",
     overViewSchedule: "Opening Hours",
     overviewFacilities: "Facilities",
-    bookingButton: "SUBMIT",
+    bookingButton: "SUMBIT",
```
Comment (ht1204) [inline src/assets/languages/languageSelector.js]: this was corrected in another PR, why is this typo here?

File: src/components/EditDetailsModal/EditDetailsModal.jsx [hunk 2]
```diff
@@ -1,44 +1,74 @@
...
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  };
 
   const handleSubmit = () => {
+    const venueId = restaurant.venues[index].id_restaurant_venue;
+    const requestBody = { field, newValue: editingItem };
+
+    axios.put(`${API_URL}/restaurant-venues/${venueId}`, requestBody, config);
```
Comment (ht1204) [inline src/components/EditDetailsModal/EditDetailsModal.jsx]: I consider async-await for this http request.

File: src/components/EditModal/EditModal.jsx [hunk 3]
```diff
@@ -1,86 +1,96 @@
...
     e.preventDefault();
     const validationErrors = {};
     const form = e.target;
     const title = form.title.value;
     const price = form.price.value;
     const description = form.description.value;
 
-    if (title.length <= 1 || typeof title !== 'string') {
-      validationErrors.title = 'title must be at least 2 characters long';
+    if (title.length <= 1 || typeof title !== "string") {
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx]: As I mentioned in another PR:
`if (!title.length || typeof title !== "string") {`

File: src/components/EditModal/EditModal.jsx [hunk 4]
```diff
@@ -1,86 +1,96 @@
...
+      validationErrors.title = "title must be at least 2 characters long";
     }
     if (price.length === 0) {
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx]: `if (!price.length) {`

File: src/components/EditModal/EditModal.jsx [hunk 5]
```diff
@@ -1,86 +1,96 @@
...
-      validationErrors.price= '* The price must be a number';
-    } else if (price <= 0){
-      validationErrors.price = 'the price cannot be zero or lower';
+      validationErrors.price = "* The price must be a number";
+    } else if (price <= 0) {
+      validationErrors.price = "the price cannot be zero or lower";
     }
-    if (description.length <= 1 || typeof description !== 'string') {
-      validationErrors.description = 'descrription must be at least 2 characters long';
+    if (description.length <= 1 || typeof description !== "string") {
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx]: `if (!description.length || typeof description !== "string") {`

File: src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx [hunk 6]
```diff
@@ -89,6 +93,84 @@ const RestaurantDetails = ({
...
+    const facilityId =
+      restaurant.venues[venueIndex].facilities[facilityIndex]
+        .id_facility_per_venue;
+
+    const response = axios.patch(
+      `${API_URL}/facilities/${facilityId}`,
+      {},
+      config
+    );
+    if (response.status === 201 || 200) {
```
Comment (ht1204) [inline src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx]: why not?
```js
const isSucceededDelete = response.status === 201 || response.status === 200;
if (isSucceededDelete) {
```

File: src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx [hunk 7]
```diff
@@ -89,6 +93,84 @@ const RestaurantDetails = ({
...
+      `${API_URL}/facilities-per-venue`,
+      {
+        facilitiesId_facility: selectedFacility,
+        restaurant_venuesId_restaurant_venue:
+          restaurant.venues[visibleVenueIndex].id_restaurant_venue,
+      },
+      config
+    );
+
+    if (response.status === 201 || 204) {
```
Comment (ht1204) [inline src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx]: `response.status === 201 || response.status === 204`

Comment (Nicolas-Rodriguez-Ch) [reply src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx]: resolved

File: src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx (hunk 6 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/pages/RestaurantAdminView/RestaurantDetails/RestaurantDetails.jsx]: resolved

File: src/components/EditModal/EditModal.jsx (hunk 5 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/EditModal/EditModal.jsx]: resolved

File: src/components/EditModal/EditModal.jsx (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/EditModal/EditModal.jsx]: Resolved

File: src/components/EditModal/EditModal.jsx (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/EditModal/EditModal.jsx]: Resolved

File: src/components/EditDetailsModal/EditDetailsModal.jsx (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/components/EditDetailsModal/EditDetailsModal.jsx]: resolved

File: src/assets/languages/languageSelector.js (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/assets/languages/languageSelector.js]: I don't know what happened, still resolved.

File: src/components/EditModal/EditModal.jsx [hunk 8]
```diff
@@ -1,86 +1,96 @@
...
   const [description, setDescription] = useState(item.description);
   const [errors, setErrors] = useState({});
-  const language = useSelector(state=> state.languageReducer);
+  const language = useSelector((state) => state.languageReducer);
 
-
-
-
-  const handleSumbit = (e) =>{
+  const handleSumbit = (e) => {
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx:13]: `handleSubmit`

File: src/components/EditModal/EditModal.jsx [hunk 9]
```diff
@@ -1,86 +1,96 @@
...
-      validationErrors.price= '* The price must be a number';
-    } else if (price <= 0){
-      validationErrors.price = 'the price cannot be zero or lower';
+      validationErrors.price = "* The price must be a number";
+    } else if (price <= 0) {
+      validationErrors.price = "the price cannot be zero or lower";
     }
-    if (description.length <= 1 || typeof description !== 'string') {
-      validationErrors.description = 'descrription must be at least 2 characters long';
+    if (description.length || typeof description !== "string") {
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx:31]: `if (!description.length || typeof description !== "string") {`

File: src/components/EditModal/EditModal.jsx [hunk 10]
```diff
@@ -1,86 +1,96 @@
...
+      validationErrors.description =
+        "descrription must be at least 2 characters long";
```
Comment (ht1204) [inline src/components/EditModal/EditModal.jsx:33]: `"description must be at least 2 characters long";`

**Pull request merged**